File: android/src/main/java/ar/com/multiplatform/biometrics/BiometricPlugin.java
package ar.com.multiplatform.biometrics;


import android.os.Handler;
import android.os.Looper;
import android.widget.Toast;
import com.getcapacitor.annotation.CapacitorPlugin;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.PluginCall;
import com.getcapacitor.Plugin;

@CapacitorPlugin(name = "Biometric")
public class BiometricPlugin extends Plugin {

    @PluginMethod
    public void echo(final PluginCall call) {
        showBiometric(call);
    }

    @PluginMethod
    public void deviceSupportBiometricLogin(final PluginCall call){
        call.resolve(Biometric.INSTANCE.generateJsonBoolean(Biometric.INSTANCE.checkIfDeviceSupportsBiometricAuth(getContext())));
    }

    public void showBiometric(final PluginCall call) {
        Handler handler = new Handler(Looper.getMainLooper());
        handler.post(() -> {
            if (Biometric.INSTANCE.checkIfDeviceSupportsBiometricAuth(getContext())) {
                Biometric.INSTANCE.showBiometricPrompt(getActivity(),null,null,null, call);
            } else {
                Biometric.INSTANCE.generateJson("Unsupported");
            }
        });
    }

}

File: android/src/main/java/ar/com/surtech/biometrics/BiometricPlugin.kt
package ar.com.surtech.biometrics

import com.getcapacitor.annotation.CapacitorPlugin
import com.getcapacitor.PluginMethod
import com.getcapacitor.PluginCall
import com.getcapacitor.JSObject
import com.getcapacitor.Plugin
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

@CapacitorPlugin(name = "Biometric")
class BiometricPlugin : Plugin() {
    val scope = CoroutineScope(Dispatchers.Main)
    private val implementation = Biometric()

    @PluginMethod
    fun echo(call: PluginCall) {
        scope.launch {
            val ret = JSObject()
            ret.put("value", showBiometric())
            call.resolve(ret)
        }
    }

    suspend fun showBiometric():String {
        return if (implementation.checkIfDeviceSupportsBiometricAuth(context)) {
            launchBiometric()
        } else {
            "UNSUPPORTED"
        }
    }

    private suspend fun launchBiometric(): String {
        return when (implementation.showBiometricPrompt(activity)) {
            is Biometric.BiometricAuthSuccess -> "OK"
            is Biometric.BiometricAuthFailed -> "FAILED"
            else -> "ERROR"
        }
    }
}